### PR TITLE
Added benchmark env info to data JSON

### PR DESCRIPTION
**Pull request by nathan-weinberg**
Also did minor reformats/additons to general NOPE code and scripts README

Example of new benchmark env data using `--jenkins-job scale-ci/e2e-benchmarking-multibranch-pipeline/kube-burner` and `--jenkins-build 491`:
```json
"benchmarkEnv": {
        "uuid": "09023530-a327-4c1c-bb23-ab877d6ee40b",
        "jenkins_job_params": [
            {
                "_class": "hudson.model.StringParameterValue",
                "name": "BUILD_NUMBER",
                "value": "108879"
            },
            {
                "_class": "hudson.model.StringParameterValue",
                "name": "WORKLOAD",
                "value": "node-density-heavy"
            },
            {
                "_class": "hudson.model.BooleanParameterValue",
                "name": "WRITE_TO_FILE",
                "value": true
            },
            {
                "_class": "hudson.model.StringParameterValue",
                "name": "JENKINS_AGENT_LABEL",
                "value": "oc411"
            },
            {
                "_class": "hudson.model.StringParameterValue",
                "name": "VARIABLE",
                "value": "200"
            },
            {
                "_class": "jenkins.plugins.parameter_separator.ParameterSeparatorValue",
                "name": "NODE_DENSITY_JOB_INFO"
            },
            {
                "_class": "hudson.model.StringParameterValue",
                "name": "NODE_COUNT",
                "value": "7"
            },
            {
                "_class": "jenkins.plugins.parameter_separator.ParameterSeparatorValue",
                "name": "CONCURRENT_BUILDS_JOB_INFO"
            },
            {
                "_class": "hudson.model.StringParameterValue",
                "name": "BUILD_LIST",
                "value": "1 8 15 30 45 60 75"
            },
            {
                "_class": "hudson.model.StringParameterValue",
                "name": "APP_LIST",
                "value": "cakephp eap django nodejs"
            },
            {
                "_class": "hudson.model.StringParameterValue",
                "name": "COMPARISON_CONFIG",
                "value": "clusterVersion.json podLatency.json podCPU-avg.json podCPU-max.json podMemory-avg.json podMemory-max.json"
            },
            {
                "_class": "hudson.model.TextParameterValue",
                "name": "ENV_VARS",
                "value": ""
            },
            {
                "_class": "hudson.model.BooleanParameterValue",
                "name": "INFRA_WORKLOAD_INSTALL",
                "value": true
            },
            {
                "_class": "hudson.model.StringParameterValue",
                "name": "SCALE_UP",
                "value": "7"
            },
            {
                "_class": "hudson.model.StringParameterValue",
                "name": "SCALE_DOWN",
                "value": "0"
            },
            {
                "_class": "hudson.model.StringParameterValue",
                "name": "E2E_BENCHMARKING_REPO",
                "value": "https://github.com/cloud-bulldozer/e2e-benchmarking"
            },
            {
                "_class": "hudson.model.StringParameterValue",
                "name": "E2E_BENCHMARKING_REPO_BRANCH",
                "value": "master"
            }
        ]
    },
```

### CODE REVIEW

**Comment by memodi** [comment]
could we remove `_class ` parameter from all jenkins job parameters? I don't think that's of any use to us.

File: scripts/nope.py [hunk 1]
```diff
@@ -170,31 +215,50 @@ def main():
...
-			print("No token could be found - ensure all the Prerequistie steps in the README were followed")
+			print("No token could be found - ensure all the Prerequisite steps in the README were followed")
 			sys.exit(1)
 	else:
-		TOKEN = subprocess.run(['oc', 'whoami', '-t'], capture_output=True, text=True).stdout
-		TOKEN = TOKEN[:-1]
-	print(f"TOKEN: {TOKEN}")
+		raw_token = subprocess.run(['oc', 'whoami', '-t'], capture_output=True, text=True).stdout
+		TOKEN = raw_token[:-1]
+	if DEBUG:
```
Comment (memodi) [inline scripts/nope.py:255]: python's logging module should help if we're using this condition all around for debug print statements.

File: scripts/README.md [hunk 2]
```diff
@@ -39,6 +39,24 @@ There are two methods you can use to install the operator:
...
+
+### Using Dittybopper
+1. Set the appropriate environmental variables by running the following:
+```bash
+export PROMETHEUS_USER_WORKLOAD_BEARER=$(oc sa get-token prometheus-user-workload -n openshift-user-workload-monitoring)
+export THANOS_URL=https://`oc get route thanos-querier -n openshift-monitoring -o json | jq -r '.spec.host'`
+```
+2. From `ocp-qe-perfscale-ci/scripts`, run `$ envsubst '${PROMETHEUS_USER_WORKLOAD_BEARER} ${THANOS_URL}' < netobserv-dittybopper.yaml.template > /tmp/netobserv-dittybopper.yaml`
+3. Clone the [performance-dashboards](https://github.com/cloud-bulldozer/performance-dashboards) repo if you haven't already
+4. From `performance-dashboards/dittybopper`, run `$ ./deploy.sh -t /tmp/netobserv-dittybopper.yaml`
```
Comment (memodi) [inline scripts/README.md:58]: add a step to import dashboard onto grafana

**Review by memodi — CHANGES_REQUESTED** [review]

File: scripts/nope.py (hunk 1 above)
Comment (nathan-weinberg) [reply scripts/nope.py:255]: i'll look into it!

File: scripts/nope.py [hunk 3]
```diff
@@ -139,19 +181,22 @@ def main():
...
 	# initialize argument parser
 	parser = argparse.ArgumentParser(description='Network Observability Prometheus and Elasticsearch tool (NOPE)')
 
 	# set customization flags
-	parser.add_argument("--yaml_file", type=str, default='netobserv-metrics.yaml', help='YAML file from which to source Prometheus queries - defaults to "netobserv-metrics.yaml"')
-	parser.add_argument("--user-workloads", default=False, action='store_true', help='Flag to query userWorkload metrics. Ensure FLP service and service-monitor are enabled and some network traffic exists.')
 	parser.add_argument("--starttime", type=str, help='Start time for range query - must be used in conjuncture with --endtime and --step')
 	parser.add_argument("--endtime", type=str, help='End time for range query - must be used in conjuncture with --starttime and --step')
 	parser.add_argument("--step", type=str, help='Step time for range query - must be used in conjuncture with --starttime and --endtime')
+	parser.add_argument("--jenkins-job", type=str, help='Jenkins job name to associate with run')
```
[the remark this answers is not in the document]
Comment (nathan-weinberg) [reply scripts/nope.py:194]: @memodi Just `scale-ci` or `scale-ci/e2e-benchmarking-multibranch-pipeline`?

[the remark this answers is not in the document]
Comment (memodi) [reply scripts/nope.py:194]: I think we can hardcode `scale-ci/e2e-benchmarking-multibranch-pipeline` and use `jenkins-job` argument to specify just the  branch job it ran with under `e2e-benchmarking-multibranch-pipeline`

**Comment by nathan-weinberg** [comment]
New example with `--jenkins-job kube-burner` and `--jenkins-build 491`:

```json
{
    "benchmarkEnv": {
        "uuid": "fef07925-d342-4879-a500-1e9c8f7e0d10",
        "jenkins_job_params": [
            {
                "name": "BUILD_NUMBER",
                "value": "108879"
            },
            {
                "name": "WORKLOAD",
                "value": "node-density-heavy"
            },
```
etc etc

File: scripts/README.md [hunk 4]
```diff
@@ -39,6 +39,25 @@ There are two methods you can use to install the operator:
 ### Setting up FLP service and creating service-monitor
 Navigate to the `scripts/` directory of this repository and run `$ populate_netobserv_metrics`
 
+### Using Dittybopper
+1. Set the appropriate environmental variables by running the following:
+```bash
+export PROMETHEUS_USER_WORKLOAD_BEARER=$(oc sa get-token prometheus-user-workload -n openshift-user-workload-monitoring)
```
Comment (paigerube14) [inline scripts/README.md:45]: Get this error when running the prometheus bearer token command 

```
 % export PROMETHEUS_USER_WORKLOAD_BEARER=$(oc sa get-token prometheus-user-workload -n openshift-user-workload-monitoring)
export THANOS_URL=https://`oc get route thanos-querier -n openshift-monitoring -o json | jq -r '.spec.host'`
error: could not find a service account token for service account "prometheus-user-workload"
```

Comment (paigerube14) [reply scripts/README.md:45]: Might need something like this line now: https://github.com/cloud-bulldozer/e2e-benchmarking/blob/64d4ca22c4395aad0b9be7230c48029207d945e8/workloads/kube-burner/common.sh#L15

Is the prometheus-user-workload sa created somewhere in the set-up scripts?

Comment (paigerube14) [reply scripts/README.md:45]: Not sure if you saw my link, but think you now have to generate a bearer token like the link above. Can OR if a token doesn't already exist

File: scripts/nope.py [hunk 5]
```diff
@@ -130,79 +174,108 @@ def main():
...
-	parser.add_argument("--user-workloads", default=False, action='store_true', help='Flag to query userWorkload metrics. Ensure FLP service and service-monitor are enabled and some network traffic exists.')
+	parser.add_argument("--debug", default=False, action='store_true', help='Flag for additional debug messaging')
 	parser.add_argument("--starttime", type=str, help='Start time for range query - must be used in conjuncture with --endtime and --step')
 	parser.add_argument("--endtime", type=str, help='End time for range query - must be used in conjuncture with --starttime and --step')
 	parser.add_argument("--step", type=str, help='Step time for range query - must be used in conjuncture with --starttime and --endtime')
-	parser.add_argument("--debug", default=False, action='store_true', help='Flag for additional debug messaging')
+	parser.add_argument("--jenkins-job", type=str, help='Jenkins job name to associate with run')
+	parser.add_argument("--jenkins-build", type=str, help='Jenkins build number to associate with run')
```
Comment (paigerube14) [inline scripts/nope.py:192]: I would try to document these nope arguments somewhere. It isn't clear from the README currently that nope has all these options

Comment (nathan-weinberg) [reply scripts/nope.py:192]: @paigerube14 maybe I could add a note saying to run `./nope.py --help` to view all command line options?

Comment (paigerube14) [reply scripts/nope.py:192]: Definitely! For the jenkins-job and jenkins-build varaibles it's not super clear on the distinction between them. And will this tool allow the option for running in not the main e2e-multibranch pipeline? Is that part of the jenkins-job parameter you pass? 
I know in jenkins there's an environment built in variable called BUILD_URL which might be useful when calling this tool from within a Jenkins job. Might be easier to just pass the full URL than separating it out into job type and build number

File: scripts/README.md (hunk 4 above)
Comment (memodi) [reply scripts/README.md:45]: interesting. Yes we should add a OR condition in the script, @nathan-weinberg - do you want to add it as part of this PR?

File: scripts/nope.py (hunk 5 above)
Comment (nathan-weinberg) [reply scripts/nope.py:192]: @paigerube14 addressing your points:
1. `jenkins-job` is the job name (e.g. `kube-burner`) and `jenkins-build` is the job's build number `e.g. 491`
2. At @memodi's request it is hardcoded to use the main e2e-multibranch pipeline - I'd like to introduce the ability to run NetObserv tests from this pipeline if possible, I'll ping you elsewhere wrt that
3. @memodi and I discussed how we wanted to pass these variables, we went with this approach as we want the tool to be able to be used locally and not just within a Jenkins env

File: scripts/README.md (hunk 4 above)
Comment (nathan-weinberg) [reply scripts/README.md:45]: Yes @memodi can do! What would the OR condition look like in our usecase?

Comment (paigerube14) [reply scripts/README.md:45]: export PROMETHEUS_USER_WORKLOAD_BEARER =$(oc sa get-token -n openshift-user-workload-monitoring prometheus-user-workload || oc sa new-token -n openshift-user-workload-monitoring prometheus-user-workload)

Comment (memodi) [reply scripts/README.md:45]: I think it should look like:
`oc sa get-token prometheus-user-workload -n openshift-user-workload-monitoring || oc sa new-token prometheus-user-workload -n openshift-user-workload-monitoring`

Comment (nathan-weinberg) [reply scripts/README.md:45]: glad you both put the same thing - that's good QE ;)

**Review by memodi — APPROVED** [review]
/lgtm

**Pull request merged**